### PR TITLE
one-line: Allow to specify "next-line"

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update

#### Overview of change:

Before, you would specify:

```json
"one-line": [true, "check-catch", "check-else", "check-finally", "check-open-brace", "check-whitespace"]
```

```json
"one-line": [true, {
	"catch": "next-line",
	"else": "next-line",
	"finally": "next-line",
	"open-brace": "same-line",
	"whitespace": true
}]
```

And get code like:

```ts
if (so) { // '{' on same line
	
}
// 'else' on next line
else {
	
}
```

#### Is there anything you'd like reviewers to focus on?

This is a breaking change to "one-line" as the options have changed.
We could instead create a new rule and deprecate "one-line". (The old name *is* less accurate now.) Or, we could deprecate the old options style but still handle it in `parseOptions`.

#### CHANGELOG.md entry:

TBD

### CODE REVIEW

**Comment by andy-hanson** [comment]
Added parsing for old-style options. Shouldn't be a breaking change any more.

File: src/rules/oneLineRule.ts [hunk 1]
```diff
@@ -20,18 +20,27 @@ import * as ts from "typescript";
...
 import * as Lint from "../index";
 
-const OPTION_BRACE = "check-open-brace";
-const OPTION_CATCH = "check-catch";
-const OPTION_ELSE = "check-else";
-const OPTION_FINALLY = "check-finally";
+const OPTION_BRACE = "open-brace";
+const OPTION_CATCH = "catch";
+const OPTION_ELSE = "else";
+const OPTION_FINALLY = "finally";
```
Comment (giladgray) [inline src/rules/oneLineRule.ts]: changing the option names is _super breaking_. do we have to do this?

File: src/rules/oneLineRule.ts [hunk 2]
```diff
@@ -20,18 +20,27 @@ import * as ts from "typescript";
...
 const OPTION_WHITESPACE = "check-whitespace";
 
+type BraceSetting = "same-line" | "next-line";
+const OPTION_SAME_LINE: BraceSetting = "same-line";
+const OPTION_NEXT_LINE: BraceSetting = "next-line";
```
Comment (giladgray) [inline src/rules/oneLineRule.ts]: swap the order, reuse the constants.

```
const OPTION_SAME_LINE = "same-line";
const OPTION_NEXT_LINE = "next-line";
type BraceSetting = OPTION_SAME_LINE | OPTION_NEXT_LINE;
```

**Comment by johnwiseheart** [comment]
Closing due to age and inactivity. Feel free to re-open or create a new pull request if you decide to continue working on this.

**Pull request closed without merging**